tools/validate-deployment.sh

```diff
@@ -0,0 +1,129 @@
+#!/bin/bash
+# This script verifies the successfull deployment of the resources needed for the DevOps OpenHack.
+# You need to provide the CSV file with all the credentials of the Azure subscriptions from the classroom management portal and a private / public SSH keypair that will be used to access the provisioning VMs
+# The error log file is where will be logged the informations regarding the failed deployments. If not provided, it defaults to error.log. 
+
+usage() { echo "Usage: validate-deployment.sh -f <errorlog_file> -u <service principal username> -x < service principal password> -t < service principal tenant>" 1>&2; exit 1; }
+
+while getopts ":f:u:x:t:" arg; do
+    case "${arg}" in
+        f)
+            ERROR_FILE=${OPTARG}
+        ;;
+        u)
+            USERNAME=${OPTARG}
+        ;;
+        x)
+            PASSWORD=${OPTARG}
+        ;;
+        t)
+            TENANT=${OPTARG}
+        ;;
+    esac
+done
+
+KEY_DIR=~/devopsohkeys
+mkdir -p ${KEY_DIR}
+ID_RSA_PRIVATE=${KEY_DIR}/devops_openhack_key
+ID_RSA_PUBLIC=${KEY_DIR}/devops_openhack_key.pub
+ssh-keygen -t rsa -C "DevOps OpenHack" -f ${ID_RSA_PRIVATE} -P ""
+
+echo "Keys have been generated"
+
+if [[ -z "$ERROR_FILE" ]]; then
+    ERROR_FILE="./errors.log"
+fi
+
+DEPLOYMENTSTATUS=0
+
+touch $ERROR_FILE
+
+az login --service-principal -u $USERNAME -p $PASSWORD --tenant $TENANT
+
+ipaddress=$(az vm list-ip-addresses --resource-group=ProctorVMG --name=proctorVM --query "[].virtualMachine.network.publicIpAddresses[].ipAddress" -otsv)
+echo IPADDRESS:$ipaddress
+
+location=$(az group show -n ProctorVMG --query location | tr -d '"')
+date=$(date '+%d/%m/%Y_%H:%M:%S')
+
+if [[ $ipaddress =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+    teamAAD=$location-$date
+    echo TEAM:$teamAAD
+    if [[ ! -d "$teamAAD" ]]; then
+    mkdir -p $teamAAD
+    fi
+    # Changing the SSH key if asked 
+    if [[ -n "$ID_RSA_PUBLIC" ]]; then
+        echo "Resetting public key to ProctorVM "
+        az vm user update --resource-group=ProctorVMG --name=proctorVM --username azureuser --ssh-key-value $ID_RSA_PUBLIC
+    else
+        echo "[ERROR] Public key missing when resetting key for ProctorVM for $teamAAD - Subscription $subid - Exiting ..."
+    fi
+
+    scp -o StrictHostKeyChecking=no -i $ID_RSA_PRIVATE -r azureuser@$ipaddress:/home/azureuser/team_env/* ./$teamAAD/
```

Comment:
$teamAAD should not be used here.

Reply:
@dcaro I am confused why it shouldnt be used here?

Reply:
Sorry, I missed that you changed the setting of the variable, this is fine then.

Reply:
@dcaro there are similar comments below too.
https://github.com/Azure-Samples/openhack-devops-proctor/pull/269#discussion_r260560233

https://github.com/Azure-Samples/openhack-devops-proctor/pull/269#discussion_r260560255